File: articles/future-directions-in-strategic-foresight/julia/foresight_capability_score.jl
# Julia standard-library example for foresight capability scoring.

profiles = Dict(
    "Government Agency" => [0.62, 0.66, 0.57, 0.63, 0.51, 0.52, 0.56, 0.58],
    "Climate Adaptation Authority" => [0.76, 0.74, 0.72, 0.70, 0.66, 0.66, 0.72, 0.68],
    "Civil Society Coalition" => [0.64, 0.60, 0.68, 0.44, 0.62, 0.84, 0.82, 0.46]
)

function foresight_capability(v)
    # v = signal scenario learning governance adaptive participation ethics data
    0.16*v[1] + 0.16*v[2] + 0.14*v[3] + 0.14*v[4] +
    0.12*v[5] + 0.10*v[6] + 0.10*v[7] + 0.08*v[8]
end

println("institution,foresight_capability")
for (name, values) in sort(collect(profiles))
    println("$(name),$(round(foresight_capability(values), digits=4))")
end

File: articles/future-directions-in-strategic-foresight/c/foresight_capability_score.c
#include <stdio.h>

double foresight_capability(double signal, double scenario, double learning, double governance, double adaptive, double participation, double ethics, double data) {
    return 0.16 * signal + 0.16 * scenario + 0.14 * learning + 0.14 * governance
        + 0.12 * adaptive + 0.10 * participation + 0.10 * ethics + 0.08 * data;
}

int main(void) {
    printf("Climate Adaptation Authority foresight_capability: %.4f\n",
           foresight_capability(0.76, 0.74, 0.72, 0.70, 0.66, 0.66, 0.72, 0.68));
    return 0;
}
